elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/Actor.kt

```diff
@@ -27,44 +27,54 @@ abstract class Actor<Command : Any, Event : Any> {
     ) = mapNotNull { eventMapper(it) }
         .catch { it.logErrorEvent(errorMapper)?.let { event -> emit(event) } ?: throw it }
 
-    protected fun <T : Any, Command : Any> Flow<T>.asSwitchFlow(command: Command, delayMillis: Long = 0): Flow<T> {
-        val key = command::class.simpleName.orEmpty()
-        return asSwitchFlow(key, delayMillis)
-    }
-
-    protected fun <T : Any> Flow<T>.asSwitchFlow(customKey: String, delayMillis: Long = 0): Flow<T> {
+    /**
+     * Extension function to switch the flow by a given key and optional delay.
+     * This function ensures that only one flow with the same key is active at a time.
+     *
+     * @param key The key to identify the flow. Defaults to the class name of the Actor.
```

Comment:
Worth to mention, that if there is more than one usage in the same Actor, don't use default key